stra.c

```diff
@@ -12,3 +12,45 @@ CHAR8 *strstra(const CHAR8 *input, const CHAR8 *find) {
 	} while (*(input++) != '\0');
 	return NULL;
 }
+
+
+/* UEFI shell populates LoadOptions and LoadOptionsSize in some odd ways.
+ * One of which is that argv[0] (the name of the efi being executed) is
+ * present in the LoadOptions.
+ *
+ * This function simply removes the first token of the command line
+ * if that token contains ends with '.efi' or '.EFI'.
+ *
+ * https://github.com/puzzleos/stubby/issues/19
+ */
+VOID remove_leading_efi_name(CHAR8 *cmdline, UINTN *len_p) {
+	CHAR8 *p, *space;
+	UINTN len = *len_p, offset = 0, newlen = 0, i = 0;
+
+	space = strstra(cmdline, (CHAR8*)" ");
+	if (space == NULL)
+		return;
+
+	p = strstra(cmdline, (CHAR8*)".EFI ");
+	if (p == NULL) {
+		p = strstra(cmdline, (CHAR8*)".efi ");
+		if (p == NULL) {
+			return;
+		}
+	}
+
+	if (p > space)
+		return;
+
+	offset = ((space + 1) - cmdline);
+	newlen = len - offset;
```

Comment:
Can you just verify that newlen > 0 here to prevent shenanigans?

Reply:
what should i do, error?
currently only returning VOID here

Reply:
Oh.  Yeah.  Is there an abort() you can do?  I mean I don't think you can safely continue if that happens.